### PR TITLE
Use Covariant123Vector DSS for Hyperdiffusion tendency

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Try dss on Covariant123Vector in hyperdiffusion tendency

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/prognostic_equations/hyperdiffusion.jl [hunk 1]
```diff
@@ -133,11 +134,14 @@ function hyperdiffusion_tendency!(Yₜ, Y, p, t)
                 # DSS on Grid scale quantities
                 # Need to split the DSS computation here, because our DSS 
                 # operations do not accept Covariant123Vector types
+                #=
                 @. ᶜ∇²uₕ = C12(ᶜ∇²u)
                 @. ᶜ∇²uᵥ = C3(ᶜ∇²u)
                 dss_op!(ᶜ∇²uₕ, buffer.ᶜ∇²uₕ)
                 dss_op!(ᶜ∇²uᵥ, buffer.ᶜ∇²uᵥ)
                 @. ᶜ∇²u = C123(ᶜ∇²uₕ) + C123(ᶜ∇²uᵥ)
+                =#
```
Comment (simonbyrne) [inline src/prognostic_equations/hyperdiffusion.jl:143]: delete this, and remove `ᶜ∇²uₕ` and `ᶜ∇²uᵥ` from the cache

File: src/prognostic_equations/hyperdiffusion.jl [hunk 2]
```diff
@@ -133,11 +134,14 @@ function hyperdiffusion_tendency!(Yₜ, Y, p, t)
...
+                dss_op!(ᶜ∇²u, buffer.ᶜ∇²u)
```
Comment (simonbyrne) [inline src/prognostic_equations/hyperdiffusion.jl:144]: pre-allocate buffer so we can reuse it?

Comment (sriharshakandala) [reply src/prognostic_equations/hyperdiffusion.jl:135]: I believe this is preallocated in `hyperdiffusion_cache`

Comment (simonbyrne) [reply src/prognostic_equations/hyperdiffusion.jl:135]: Oh, i misread it.

**Comment by simonbyrne** [comment]
Looks like an assertion is failing:
```
AssertionError: S <: supportedvectortypes
  Stacktrace:
    [1] create_dss_buffer(data::ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.Covariant123Vector{Float32}, 4, Array{Float32, 5}}, topology::ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float32}, ClimaCore.Meshes.NormalizedBilinearMap}, Vector{CartesianIndex{3}}, Array{Int64, 3}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, Vector{Tuple{Int64, Int64}}, Vector{Int64}, Vector{Tuple{Bool, Int64, Int64}}, Vector{Int64}, NamedTuple{(), Tuple{}}, Vector{Tuple{Int64, Int64}}}, perimeter::ClimaCore.Spaces.Perimeter2D{4}, local_geometry::ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.LatLongPoint{Float32}, Float32, StaticArraysCore.SMatrix{2, 2, Float32, 4}}, 4, Array{Float32, 4}}, local_weights::ClimaCore.DataLayouts.IJFH{Float32, 4, Array{Float32, 4}})
```

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by simonbyrne** [comment]
@sriharshakandala you don't need to `bors try` on ClimaAtmos, all tests run by default.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**